ui/src/common/ScrollUpButton.tsx

```diff
@@ -3,11 +3,43 @@ import KeyboardArrowUp from '@material-ui/icons/KeyboardArrowUp';
 import React, {Component} from 'react';
 
 class ScrollUpButton extends Component {
+    state = {
+        display: 'none',
+        opacity: 0,
+    };
+    componentDidMount() {
+        if (typeof window !== 'undefined') {
+            window.addEventListener('scroll', this.scrollHandler.bind(this));
+        }
+    }
+
+    componentWillUnmount() {
+        if (typeof window !== 'undefined') {
+            window.removeEventListener('scroll', this.scrollHandler.bind(this));
```

Comment:
I think this doesn't work as .bind will give a new reference and thus, doesn't unregister the old handler. I'll fix this. Also, I'll prevent some rerenders, by checking if the state will be changed.